src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -77,7 +77,8 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
 
     public async loadMoreChildrenImpl(_clearCache: boolean): Promise<AzExtTreeItem[]> {
         let dbNames: (string | undefined)[];
-        if (this._azureId) {
+        if (this.azureName) {
+            await this.getCredentials();
```

Comment:
How come you added getCredentials here?

Reply:
Adding `getCredentials()` here would make sure that the `connectionString.username` and `connectionString.password` are initialized. Because the connectionString is not stored in persistedId like username every time loadMore...() runs the connectionString is reset and the username and password are undefined again. 

With `getCredentials()` here username and password values of the conenctionString are available to use in the `getClientConfig()` ( do not have to call `getCredentials()` again in `getClientConfig()`)

Reply:
I see what you're trying to fix, but I'm worried about two things:
1. There's no guarantee "loadMoreChildrenImpl" is called before connectionString is accessed, so we might run into timing issues
1. getCredentials accesses keytar every time its called, and we want to minimize the number of times we use keytar. One reason is that operating systems will prompt users something like "Do you authorize VS Code to access keychain?". Users can usually click "Always Allow", but we don't want to bother them if they don't want to click that.

I think I prefer if fixing this issue is done separately, as I don't think it'll be a one-line fix like you have here